File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/C++/_029)_Maximum_Circular_Subarray_Sum.cpp
#include<iostream>
using namespace std;

int maxCircularSubarraySum(int arr[], int n)
{
    int res = arr[0];
    for(int i=0; i<n; i++)
    {
        int curr_max = arr[i];
        int curr_sum = arr[i];
        for(int j=1; j<n; j++)
        {
            int index = (i+j)%n;
            curr_sum += arr[index];
            curr_max = max(curr_max, curr_sum);
        }
        res = max(res, curr_max);
    }
    return res;
}

int main()
{
    int arr[] = {5, -2, 3, 4}, n=4;
    cout<<"Maximum Circular SubArray Sum = "<<maxCircularSubarraySum(arr,n)<<endl;
}

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_029)_Maximum_Circular_Subarray_Sum.dart
import 'dart:math';

int maxCircularSubArraySum(List<int> arr) {
  int res = arr[0];

  for (int i = 0; i < arr.length; i++) {
    int currMax = arr[i];
    int currSum = arr[i];

    for (int j = 1; j < arr.length; j++) {
      int index = (i + j) % arr.length;
      currSum += arr[index];
      currMax = max(currMax, currSum);
    }
    res = max(res, currMax);
  }
  return res;
}

void main() {
  var arr = [5, -2, 3, 4];
  print('Maximum circular sub-array sum is ${maxCircularSubArraySum(arr)}.');
}
